src/approaches/grammar_search_invention_approach.py

```diff
@@ -647,6 +647,7 @@ class _HeuristicBasedScoreFunction(_OperatorLearningBasedScoreFunction):
     against the demonstrations.
     """
     heuristic_names: Sequence[str]
+    demos_only: bool = field(default=True)
```

Comment:
are we planning to make a CFG flag for this later?

Reply:
I'm planning to make a different score function that uses `demos_only=False` later

Reply:
that would be a subclass of this one?

Reply:
yeah well this would be a mixin, like we do for the existing lookahead score functions